jax/_src/numpy/lax_numpy.py

```diff
@@ -3246,7 +3246,8 @@ def _einsum(
     preferred_element_type,
     _dot_general=lax.dot_general,
 ):
-  operands = list(util.promote_dtypes(*operands))
+  if preferred_element_type is None:
+    operands = list(util.promote_dtypes(*operands))
```

Comment:
Will this do the right thing in the cases of einsum that do not bottom out in calling `dot()`?

Reply:
![image](https://github.com/google/jax/assets/1458824/9517b814-03d6-4d89-933e-36be5fdca35a)